CHANGELOG.md

```diff
@@ -1,3 +1,197 @@
+
+This is a rollup of all of the bugfixes merged, and features introduced to RDFLib since 
+RDFLib 4.2.2 was released in Jan 2017.
+
+While all effort was taken to minimize breaking changes in this release, there are some.
+
+Please see the upgrade4to5 document for more information on some specific differences from 4.2.2 to 5.0.0.
+
+_**All issues closed and PRs merged since 4.2.2:**_
+
+### General Bugs Fixed:
+  * Pr 451 redux
+    [#978](https://github.com/RDFLib/rdflib/pull/978)
+  * Issue 920 fixes as per Issue desc + test
+    [#974](https://github.com/RDFLib/rdflib/pull/974)
+  * Remove colons from test result files. Fix #901.
+    [#971](https://github.com/RDFLib/rdflib/pull/971)  
+  * Add requirement for requests to setup.py
+    [#969](https://github.com/RDFLib/rdflib/pull/969)
+  * fixed URIRef including native unicode characters
+    [#961](https://github.com/RDFLib/rdflib/pull/961)
+  * DCTERMS.format not working
+    [#932](https://github.com/RDFLib/rdflib/issues/932)
+  * NTriples fails to parse URIs with only a scheme
+    [#920](https://github.com/RDFLib/rdflib/issues/920)
+  * infixowl.manchesterSyntax do not encode strings
+    [#906](https://github.com/RDFLib/rdflib/pull/906)
+  * cannot clone it on windows
+    [#901](https://github.com/RDFLib/rdflib/issues/901)
+  * Fix blank node label to not contain '_:' during parsing
+    [#886](https://github.com/RDFLib/rdflib/pull/886)
+  * rename new SPARQLWrapper to SPARQLConnector
+    [#872](https://github.com/RDFLib/rdflib/pull/872)
+  * Fix #859. Unquote and Uriquote Literal Datatype.
+    [#860](https://github.com/RDFLib/rdflib/pull/860)
+  * Parsing nquads
+    [#786](https://github.com/RDFLib/rdflib/issues/786)
+  * ntriples spec allows for upper-cased lang tag, fixes #782
+    [#784](https://github.com/RDFLib/rdflib/pull/784)
+  * Error parsing N-Triple file using RDFlib
+    [#782](https://github.com/RDFLib/rdflib/issues/782)
+  * Adds escaped single quote to literal parser
+    [#736](https://github.com/RDFLib/rdflib/pull/736)
+  * N3 parse error on single quote within single quotes 
+    [#732](https://github.com/RDFLib/rdflib/issues/732)
+  * Fixed #725 
+    [#730](https://github.com/RDFLib/rdflib/pull/730)
+  * test for issue #725: canonicalization collapses BNodes
+    [#726](https://github.com/RDFLib/rdflib/pull/726)
+  * RGDA1 graph canonicalization sometimes still collapses distinct BNodes
+    [#725](https://github.com/RDFLib/rdflib/issues/725)
+  * Accept header should use a q parameter
+    [#720](https://github.com/RDFLib/rdflib/pull/720)
+  * Added test for Issue #682 and fixed.
+    [#718](https://github.com/RDFLib/rdflib/pull/718)
+  * Incompatibility with Python3: unichr
+    [#687](https://github.com/RDFLib/rdflib/issues/687)
+  * namespace.py include colon in ALLOWED_NAME_CHARS
+    [#663](https://github.com/RDFLib/rdflib/pull/663)
+  * Parse implicit string
+    [#657](https://github.com/RDFLib/rdflib/pull/657)
```

Comment:
#657 was closed without being merged, should it be on this list?

Reply:
The script we're using to generate this changelog lists everything that was closed since the last version and isn't specific on how (closed, merged etc.). We're not thrilled about this and @ashleysommer was looking at including next to each item labels to indicate *how* it was closed but that's not come through just yet.

Perhaps here it would be good to see:

  * PR: Parse implicit string
    [#657](https://github.com/RDFLib/rdflib/pull/657) 
      * Outcome: *Closed*

As opposed to:

  * PR Parse implicit string
    [#657](https://github.com/RDFLib/rdflib/pull/657) 
      * Outcome: *Merged*

@ashleysommer: could the marker "PR:" be added for for all PRs and "Issue:" for all Issues, given that the URI with /pull/ or /issues/ in it is hidden form rendering (yeah, mouseover would show it...) and could Merged/Closed be indicated for PRs? I guess all Issues are Closed.

Reply:
In this new set of changes, I've added ISSUE: or PR: against the number.

Adding outcomes would require significant refactoring of the Changelog generation script. Its certainly possible via the Github API to get these, but I want to rewrite the script in Python to add these custom API features. Will need to wait until next release to see that.